test/evergreen.yml

```diff
@@ -1937,11 +1937,33 @@ tasks:
     commands:
       - func: "get project"
       - func: "compile wiredtiger address sanitizer"
-      - func: "format test script"
-        vars:
-          test_env_vars: ASAN_OPTIONS="detect_leaks=1:abort_on_error=1:disable_coredump=0" ASAN_SYMBOLIZER_PATH=/opt/mongodbtoolchain/v3/bin/llvm-symbolizer
-          # run for 24 hours ( 24 * 60 = 1440 minutes), don't stop at failed tests, use default config
-          format_test_script_args: -t 1440
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger/test/format"
+          shell: bash
+          script: |
+            set -o verbose
+
+            ASAN_OPTIONS="detect_leaks=1:abort_on_error=1:disable_coredump=0" ASAN_SYMBOLIZER_PATH=/opt/mongodbtoolchain/v3/bin/llvm-symbolizer
+
+            # run for 24 hours ( 24 * 60 = 1440 minutes), don't stop at failed tests, use default config
+            ${test_env_vars|} ${format_test_setting|} ./format.sh ${smp_command|} -t 1440 2>&1
+            format_ret=$?
+
+            # If any of the format job fails i.e, format.sh script returns 1 then check if it failed with
+            # any of the known errors.
+            if [ $format_ret -ne 0 ]
+            then
+              # Call the generic format skip error script to skip known errors
+              ./format_skip_errors.sh
+              if [ $? -eq 0 ]
+              then
+                exit 0
+              fi
+            fi
+
+            # Propogate the exit status of format script.
+            exit $format_ret
```

Comment:
Is it possible to make a function variable for `format test script` to control whether or not performing further "known error skipping" logic and abstract this part of code into function `format test script`? This way we can easily reuse the code in other format tasks.

Reply:
I will include the "skip known errors" logic in the format.sh script and can be reused by all the existing tasks.